Edge Storage Accelerator (ESA) - Jumpstart Scenario (#277)
Azure/arc_jumpstart_docs: docs/azure_arc_jumpstart/azure_edge_iot_ops/aks_edge_essentials_single_esa/_index.md

---
type: docs

> title: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"

Reviewer: ```suggestion
title: "Edge Storage Accelerator on AKS Edge Essentials single node deployment"
```

> linkTitle: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"

Reviewer: ```suggestion
linkTitle: "Edge Storage Accelerator on AKS Edge Essentials single node deployment"
```

> weight: 4

Reviewer: ```suggestion
weight: 8
```

description: >
---

> ## Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment 

Reviewer: ```suggestion
## Edge Storage Accelerator on AKS Edge Essentials single node deployment 
```

> The following Jumpstart scenario showcases Edge Storage Accelerator (ESA), which is a storage system designed for Arc-enabled Kubernetes clusters to provide reliable, fault tolerant storage in a ReadWriteMany persistent volume. The Edge Storage Accelerator provides a constantly connected conduit for edge data to be replicated in cloud on blob while maintaining a local copy, as space permits, for low latency local access. In this scenario, a model detects defects in bolts by analyzing video from a supply line streamed over RTSP. The identified defects are then stored in a container within a storage account using ESA.

Reviewer: ```suggestion
The following Jumpstart scenario showcases Edge Storage Accelerator (ESA), which is a storage system designed for Arc-enabled Kubernetes clusters to provide reliable, fault tolerant storage in a ReadWriteMany persistent volume. The Edge Storage Accelerator provides a constantly connected conduit for edge data to be replicated to blob storage in the cloud while maintaining a local copy, as space permits, for low-latency local access. 

In this scenario, a computer vision AI model detects defects in bolts by analyzing video from a supply line video feed streamed over RTSP. The identified defects are then stored in a container within a storage account using ESA.
```

> This scenario will help you create deploy Edge Storage Accelerator by creating an AKS Edge Essentials single node deployment in an Azure Windows Server VM. The provided ARM template is responsible for creating the Azure resources as well as executing the LogonScript (AKS Edge Essentials cluster creation, Azure Arc onboarding (Azure VM and AKS Edge Essentials cluster) and Edge Storage Accelerator deployment) on the Azure VM. Once Edge Essentials is deployed [Edge Storage Accelerator](https://learn.microsoft.com/azure/azure-arc/edge-storage-accelerator/overview) is installed as a Kubernetes service that exposes a CSI driven storage class for use by applications in the Edge Essentials Kubernetes cluster.

Reviewer: ```suggestion
The automation in this scenario deploys Edge Storage Accelerator on an AKS Edge Essentials single-node deployment running in an Azure virtual machine. The provided ARM template creates the Azure resources and configures the LogonScript.ps1 custom script extension which handles AKS Edge Essentials cluster creation, Azure Arc onboarding for the Azure VM and AKS Edge Essentials cluster, and Edge Storage Accelerator deployment. Once AKS Edge Essentials is deployed [Edge Storage Accelerator](https://learn.microsoft.com/azure/azure-arc/edge-storage-accelerator/overview) is installed as a Kubernetes service that exposes a CSI driven storage class for use by applications in the Edge Essentials Kubernetes cluster.
```

    > **Note:** The script run time is ~13min long.

    ![Screenshot script output](./12.png)

    ![Screenshot script output](./13.png)

    ![Screenshot script output](./14.png)

    ![Screenshot script output](./15.png)

    ![Screenshot script output](./16.png)

    ![Screenshot script output](./17.png)

    ![Screenshot script output](./18.png)

    ![Screenshot script output](./19.png)

    ![Screenshot script output](./38.png)

- Upon successful run, a new Azure Arc-enabled server and Azure Arc-enabled Kubernetes cluster will be added to the resource group.

![Screenshot Azure Arc-enabled server on resource group](./22.png)

- You can also run _kubectl get nodes -o wide_ to check the cluster node status and _kubectl get pod -A_ to see that the cluster is running and all the needed pods (system, [the Arc-enabled Kubernetes extension pods](https://learn.microsoft.com/azure/azure-arc/kubernetes/extensions), and [Azure Monitor extension pods](https://learn.microsoft.com/azure/azure-monitor/containers/container-insights-overview) are in a running state.

![Screenshot kubectl get nodes -o wide](./36.png)

![Screenshot kubectl get pod -A](./37.png)

> 

Reviewer: remove extra blank line

---
type: docs
title: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"
linkTitle: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"
weight: 4
description: >
---

## Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment 

> The following Jumpstart scenario will show how to create an AKS Edge Essentials single node deployment in an Azure Windows Server VM and connect the Azure VM and AKS Edge Essentials cluster to Azure Arc. The provided ARM template is responsible for creating the Azure resources as well as executing the LogonScript (AKS Edge Essentials cluster creation, Azure Arc onboarding (Azure VM and AKS Edge Essentials cluster) and Edge Storage Accelerator deployment) on the Azure VM. Once Edge Essentials is deployed [Edge Storage Accelerator](https://learn.microsoft.com/en-us/azure/azure-arc/edge-storage-accelerator/overview) is installed as a Kubernetes service that exposes a CSI driven storage class for use by applications in the Edge Essentials Kubernetes cluster.

Reviewer 2: - remove en-us from all links
- Make sure you have a call out that ESA is in preview. Consult with the team for examples

Azure VMs leverage the [Azure Instance Metadata Service (IMDS)](https://learn.microsoft.com/azure/virtual-machines/windows/instance-metadata-service) by default to provide information about the VMs, and to manage and configure them. By projecting an Azure VM as an Azure Arc-enabled server, a "conflict" is created which will not allow the Azure Arc server resources to be represented as one resource when the IMDS is being used. Instead, the Azure Arc server will still "act" as a native Azure VM.

However, **for demo purposes only**, the below guide will allow you to use and onboard Azure VMs to Azure Arc. By doing so, you will be able to simulate a server that is deployed outside of Azure (i.e. "on-premises" or in other cloud platforms)

> **Note:** It is not expected for an Azure VM to be projected as an Azure Arc-enabled server. The below scenario is unsupported and should ONLY be used for demo and testing purposes.

> ![Screenshot ARM template output](./01.png)

Reviewer 2: this is not the scenario architecture diagram. we created a dedicated scenario diagram as well ESA architecture diagram that you can consider including. /cc @dkirby-ms

---
type: docs
title: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"
linkTitle: "Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment"
weight: 4
description: >
---

## Discover Edge Storage Accelerator on AKS Edge Essentials single node deployment 

The following Jumpstart scenario will show how to create an AKS Edge Essentials single node deployment in an Azure Windows Server VM and connect the Azure VM and AKS Edge Essentials cluster to Azure Arc. The provided ARM template is responsible for creating the Azure resources as well as executing the LogonScript (AKS Edge Essentials cluster creation, Azure Arc onboarding (Azure VM and AKS Edge Essentials cluster) and Edge Storage Accelerator deployment) on the Azure VM. Once Edge Essentials is deployed [Edge Storage Accelerator](https://learn.microsoft.com/en-us/azure/azure-arc/edge-storage-accelerator/overview) is installed as a Kubernetes service that exposes a CSI driven storage class for use by applications in the Edge Essentials Kubernetes cluster.

> Edge Storage Accelerator (ESA) is a storage system designed for Arc-connected Kubernetes clusters to provide reliable, fault tolerant storage in a ReadWriteMany persistent volume. The Edge Storage Accelerator provides a constantly connected conduit for edge data to be replicated in cloud on blob while maintaining a local copy, as space permits, for low latency local access. 

Reviewer 3: ```suggestion
Edge Storage Accelerator (ESA) is a storage system designed for Arc-enabled Kubernetes clusters to provide reliable, fault tolerant storage in a ReadWriteMany persistent volume. The Edge Storage Accelerator provides a constantly connected conduit for edge data to be replicated in cloud on blob while maintaining a local copy, as space permits, for low latency local access. 
```

    ```json
    {
    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "displayName": "JumpstartArc",
    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
    ```

    > **Note:** If you create multiple subsequent role assignments on the same service principal, your client secret (password) will be destroyed and recreated each time. Therefore, make sure you grab the correct password.

    > **Note:** The Jumpstart scenarios are designed with ease of use in-mind and adhere to security-related best practices whenever possible. It is optional but highly recommended to scope the service principal to a specific [Azure subscription and resource group](https://learn.microsoft.com/cli/azure/ad/sp?view=azure-cli-latest) as well as considering use of a [less privileged service principal account](https://learn.microsoft.com/azure/role-based-access-control/best-practices).

- Create a blob storage account
    Edge Storage Accelerator leverages a blob storage account as the final destination of data from the edge. You must create a blob storage account with or without Hierarcial Name Space (HNS) support in order to use ESA.

>     ```shell

Reviewer 3: Why don't we create the storage account in the automation ?

Author: We do! This section is outdated, I will change it now.

    az storage account create --name uniqueAccountName --resource-group myStorageRG
    az storage container create --name esaStorageContainer
    ```

## Automation Flow

For you to get familiar with the automation and deployment flow, below is an explanation.

- User edits the ARM template parameters file (1-time edit). These parameter values are used throughout the deployment.

> - Main [_azuredeploy_ ARM template](https://github.com/microsoft/azure_arc/blob/main/azure_arc_k8s_jumpstart/aks_hybrid/aks_edge_essentials_single_akri/arm_template/azuredeploy.json) will initiate the deployment of the following resources:

Reviewer 3: Is it expected that the link points to the Akri scenario code ?

Author: fixed

  - _Virtual Network_ - Virtual Network for Azure Windows Server VM.
  - _Network Interface_ - Network Interface for Azure Windows Server VM.
  - _Network Security Group_ - Network Security Group to allow RDP in Azure Windows Server VM.
  - _Virtual Machine_ - Azure Windows Server VM.
  - _Custom script and Azure Desired State Configuration extensions_ - Configure the Azure Windows Server VM to host AKS Edge Essentials.

> - User remotes into client Windows VM, which automatically kicks off the [_LogonScript_](https://github.com/microsoft/azure_arc/blob/main/azure_arc_k8s_jumpstart/aks_hybrid/aks_edge_essentials_single_akri/arm_template/artifacts/LogonScript.ps1) PowerShell script to:

Reviewer 3: Is it expected that the link points to the Akri scenario code ?

Author: updated
